thumbor/handlers/__init__.py

```diff
@@ -485,57 +485,45 @@ def _fetch(self, url):
         :rtype:
         """
         fetch_result = FetchResult()
+        loader_result = None
 
         storage = self.context.modules.storage
 
         yield self.acquire_url_lock(url)
 
         try:
             fetch_result.buffer = yield gen.maybe_future(storage.get(url))
-            mime = None
 
             if fetch_result.buffer is not None:
                 self.release_url_lock(url)
-
                 fetch_result.successful = True
-
                 self.context.metrics.incr('storage.hit')
-                mime = BaseEngine.get_mimetype(fetch_result.buffer)
-                self.context.request.extension = EXTENSION.get(mime, '.jpg')
-                if mime == 'image/gif' and self.context.config.USE_GIFSICLE_ENGINE:
-                    self.context.request.engine = self.context.modules.gif_engine
-                else:
-                    self.context.request.engine = self.context.modules.engine
-
-                raise gen.Return(fetch_result)
```

Comment:
this block is gone completely and more down below a similar part is doing almost the same but also `engine.load()` the image and has additional checks. Is this the fix hear (regarding MAX_PIXELS)? If not, then this would at least change the behavior quiet a lot (which doesn't mean that it is the right thing to do, actually I have some problems overseeing all the affected parts)